instat/dlgAddLink.vb

```diff
@@ -35,28 +37,108 @@ Public Class dlgAddLink
 
     Private Sub InitialiseDialog()
         ucrBase.iHelpTopicID = 506
+        ucrDataSelectorFrom.SetParameter(New RParameter("from_data_frame", 0))
+        ucrDataSelectorFrom.SetParameterIsString()
+        ucrDataSelectorTo.SetParameter(New RParameter("to_data_frame", 0))
+        ucrDataSelectorTo.SetParameterIsString()
+        ucrInputLinkName.SetParameter(New RParameter("link_name", 0))
+        lvwLinkViewBox.Columns.Add("Name", 80, HorizontalAlignment.Left)
+        lvwLinkViewBox.Columns.Add("Columns", 150, HorizontalAlignment.Left)
+        ucrInputSelectedKey.IsReadOnly = True
     End Sub
 
     Private Sub SetDefaults()
-        Dim clsDefaultFunction As New RFunction
-
-        'clsDefaultFunction.SetRCommand()
-        'ucrBase.clsRsyntax.SetBaseRFunction(clsDefaultFunction)
+        clsAddLink = New RFunction
+        ucrDataSelectorFrom.Reset()
+        ucrDataSelectorTo.Reset()
+        UpdateKeys()
+        clsAddLink.AddParameter("type", Chr(34) & "keyed_link" & Chr(34))
+        clsAddLink.SetRCommand(frmMain.clsRLink.strInstatDataObject & "$add_link")
+        ucrBase.clsRsyntax.SetBaseRFunction(clsAddLink)
+        cmdSpecifyLink.Enabled = False
     End Sub
 
     Private Sub SetRCodeForControls(bReset As Boolean)
         SetRCode(Me, ucrBase.clsRsyntax.clsBaseFunction, bReset)
     End Sub
 
     Private Sub TestOKEnabled()
-    End Sub
-
-    Private Sub ReopenDialog()
+        If ucrDataSelectorFrom.cboAvailableDataFrames.Text <> "" AndAlso ucrDataSelectorTo.cboAvailableDataFrames.Text <> "" AndAlso Not ucrInputLinkName.IsEmpty AndAlso Not ucrInputSelectedKey.IsEmpty Then
+            ucrBase.OKEnabled(True)
+        Else
+            ucrBase.OKEnabled(False)
+        End If
     End Sub
 
     Private Sub ucrBaseReplace_ClickReset(sender As Object, e As EventArgs) Handles ucrBase.ClickReset
         SetDefaults()
         SetRCodeForControls(True)
         TestOKEnabled()
     End Sub
+
+    Private Sub ucrDataSelectorTo_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrDataSelectorTo.ControlValueChanged
+        UpdateKeys()
+        TestOKEnabled()

```

Comment:
Should we call TestOk in this sub or include  "ucrDataSelectorTo" in the contentsChanged events of TestOk below? Well, may be you have a good reason..